apps/dhparam.c

```diff
@@ -371,17 +371,9 @@ int dhparam_main(int argc, char **argv)
 
 static int dh_cb(int p, int n, BN_GENCB *cb)
 {
-    char c = '*';
-
-    if (p == 0)
-        c = '.';
-    if (p == 1)
-        c = '+';
-    if (p == 2)
-        c = '*';
-    if (p == 3)
-        c = '\n';
-    BIO_write(BN_GENCB_get_arg(cb), &c, 1);
+    static const char symbols[] = ".+*\n";
```

Comment:
I like this approach but it is broken.  What happens if **p** is not a member of the set _{ 0, 1, 2, 3 }_?  The old code is deterministic in this case.

I'm aware that the existing calls pass always 0 - 3 as this argument.  But there is no reason to not be defensive.

Reply:
So you'd be fine with a range check?

Reply:
Yes.